subsys/net/lib/nrf_cloud/coap/src/nrf_cloud_coap_transport.c

```diff
@@ -50,12 +50,23 @@ LOG_MODULE_REGISTER(nrf_cloud_coap_transport, CONFIG_NRF_CLOUD_COAP_LOG_LEVEL);
 #define VER_STRING_FMT2 "cver=" CDDL_VERSION "&dver=" BUILD_VERSION_STR
 #define BUILD_VERSION_STR STRINGIFY(BUILD_VERSION)
 
-static int sock = -1;
-static bool authenticated;
-static bool cid_saved;
+#define NRF_CLOUD_COAP_AUTH_RSC "auth/jwt"
 
-static struct coap_client coap_client;
-static int nrf_cloud_coap_authenticate(void);
+/* CoAP client transfer data */
+struct cc_xfer_data {
+	struct nrf_cloud_coap_client *const nrfc_cc;
+	coap_client_response_cb_t cb;
+	void *user_data;
+	int result_code;
+	struct k_sem *sem;
+};
+
+/* Semaphore to be used with internal coap_client requests */
+static K_SEM_DEFINE(cb_sem, 0, 1);
+/* Semaphore to be used when doing authorization with an external coap_client */
+static K_SEM_DEFINE(ext_cc_sem, 0, 1);
```

Comment:
Would it make more sense for `ext_cc_sem` to be a part of the nrf_cloud_coap_client struct?

Reply:
this is just used for auth and, unless things change, there will only be one external coap client (for downloads).